app/assets/javascripts/discourse/tests/unit/lib/autocomplete-test.js

```diff
@@ -0,0 +1,210 @@
+import { module, test } from "qunit";
+import autocomplete from "discourse/lib/autocomplete";
+import { compile } from "handlebars";
+
+module("Unit | Utility | autocomplete", function (hooks) {
```

Comment:
I am impressed you've made this all work in a unit spec, personally I would have done this in a system one now since they are easier to write, **however** what we don't get with those is an easy way to do what you've done here, make a textarea specifically for testing this one feature. Just a thought, but maybe we would benefit from a kind of playground route for system specs that loads with a textbox, textarea, and a select menu where JS can be injected and run then a specific system spec is run for library/component functionality like this.

Probably overkill, but something to think about...

Reply:
yeah system tests may be an interesting use here... for sure... I do though love how super fast this test is... that is one advantage of the unit approach. Disadvantage though is all the complex event handling, it is not trivial to test.